compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -1081,10 +1082,12 @@ public void visit(BLangAnnotation annotationNode) {
 
     public void visit(BLangAnnotationAttachment annAttachmentNode) {
         annAttachmentNode.expr = rewrite(annAttachmentNode.expr, env);
-        // TODO: need to check this. Balo creation in java module fails in java11 migration.
-//        if (annAttachmentNode.expr != null) {
-//            annAttachmentNode.expr = visitCloneReadonly(annAttachmentNode.expr, annAttachmentNode.expr.type);
-//        }
+        for (AttachPoint point : annAttachmentNode.annotationSymbol.points) {
+            if (annAttachmentNode.expr != null && !point.source) {
+                annAttachmentNode.expr = visitCloneReadonly(annAttachmentNode.expr, annAttachmentNode.expr.type);
+                break;
+            }
+        }
```

Comment:
We don't need to iterate through the points if `annAttachmentNode.expr` is null, right? 

Shall we refactor this to check that first?

```suggestion
        if (annAttachmentNode.expr != null) {
            for (AttachPoint point : annAttachmentNode.annotationSymbol.points) {
                if (!point.source) {
                    annAttachmentNode.expr = visitCloneReadonly(annAttachmentNode.expr, annAttachmentNode.expr.type);
                    break;
                }
            }
        }
```

Reply:
Yes. Changed.